num fails to build with dune 3.0

Building with dune 2.x never failed AFAIR. But with 3.0 it fails consistently:

````
[   14s] + dune installed-libraries
[   14s] bigarray               (version: [distributed with OCaml])
[   14s] bytes                  (version: [distributed with OCaml])
[   14s] compiler-libs          (version: [distributed with OCaml])
[   14s] compiler-libs.bytecomp (version: [distributed with OCaml])
[   14s] compiler-libs.common   (version: [distributed with OCaml])
[   14s] compiler-libs.optcomp  (version: [distributed with OCaml])
[   14s] compiler-libs.toplevel (version: [distributed with OCaml])
[   14s] dynlink                (version: [distributed with OCaml])
[   14s] ocamldoc               (version: [distributed with OCaml])
[   14s] raw_spacetime          (version: 4.13.1)
[   14s] seq                    (version: 4.13.1)
[   14s] stdlib                 (version: [distributed with OCaml])
[   14s] str                    (version: [distributed with OCaml])
[   14s] threads                (version: [distributed with OCaml])
[   14s] threads.posix          (version: [distributed with OCaml])
[   14s] uchar                  (version: 4.13.1)
[   14s] unix                   (version: [distributed with OCaml])
...
[   14s] + dune build --verbose --for-release-of-packages=num -j8 @install
...
[   14s] Actual targets:
[   14s] - recursive alias @install
[   15s] Running[1]: (cd _build/default/src && /usr/bin/gcc-11 -O2 -fno-strict-aliasing -fwrapv -pthread -Werror=implicit-function-declaration -Werror=return-type -Wno-deprecated-declarations -ffat-lto-objects -fPIC -pipe -D_FILE_OFFSET_BITS=64 -DBNG_ARCH_amd64 -g -I /usr/lib64/ocaml -o bng.o -c bng.c)
[   15s] Running[2]: (cd _build/default/src && /usr/bin/gcc-11 -O2 -fno-strict-aliasing -fwrapv -pthread -Werror=implicit-function-declaration -Werror=return-type -Wno-deprecated-declarations -ffat-lto-objects -fPIC -pipe -D_FILE_OFFSET_BITS=64 -DBNG_ARCH_amd64 -g -I /usr/lib64/ocaml -o nat_stubs.o -c nat_stubs.c)
[   15s] Running[3]: (cd _build/default && /usr/bin/ocamlc.opt -w -40 -g -a -o src/num.cma)
[   15s] Running[4]: (cd _build/default && /usr/bin/ocamlopt.opt -w -40 -g -a -o src/num.cmxa)
[   15s] Running[5]: (cd _build/default && /usr/bin/ocamlopt.opt -w -40 -g -shared -linkall -I src -o src/num.cmxs src/num.cmxa)
[   15s] File "src/dune", line 14, characters 21-24:
[   15s] 14 |   (c_names nat_stubs bng bng_generic)
[   15s]                           ^^^
[   15s] Command [1] exited with code 1:
[   15s] $ (cd _build/default/src && /usr/bin/gcc-11 -O2 -fno-strict-aliasing -fwrapv -pthread -Werror=implicit-function-declaration -Werror=return-type -Wno-deprecated-declarations -ffat-lto-objects -fPIC -pipe -D_FILE_OFFSET_BITS=64 -DBNG_ARCH_amd64 -g -I /usr/lib64/ocaml -o bng.o -c bng.c)
[   15s] bng.c:23:10: fatal error: bng_amd64.c: No such file or directory
[   15s]    23 | #include "bng_amd64.c"
[   15s]       |          ^~~~~~~~~~~~~
[   15s] compilation terminated.
[   15s] Running[6]: (cd _build/default/src && /usr/bin/touch bng_generic.c)
````

--- a/dune-project
+++ b/dune-project
@@ -1,1 +1,1 @@
-(lang dune 1.10)
+(lang dune 2.0)

--- a/src/dune
+++ b/src/dune
@@ -11,8 +11,11 @@
   (public_name num.core)
   (wrapped false)
   (modules arith_flags arith_status big_int int_misc nat num ratio)
-  (c_names nat_stubs bng bng_generic)
-  (c_flags "-DBNG_ARCH_%{architecture}")
+  (foreign_stubs
+   (language c)
+   (names nat_stubs bng bng_generic)
+   (extra_deps (glob_files *.c))
+   (flags "-DBNG_ARCH_%{architecture}"))
   (flags -w +a-4-9-41-42-44-45-48 -warn-error +A -bin-annot -g -safe-string -strict-sequence -strict-formats))
 
 (rule

--- a/src/dune
+++ b/src/dune
@@ -13,12 +13,7 @@
   (modules arith_flags arith_status big_int int_misc nat num ratio)
   (foreign_stubs
    (language c)
-   (names nat_stubs bng bng_generic)
+   (names nat_stubs bng)
    (extra_deps (glob_files *.c))
    (flags "-DBNG_ARCH_%{architecture}"))
-  (flags -w +a-4-9-41-42-44-45-48 -warn-error +A -bin-annot -g -safe-string -strict-sequence -strict-formats))
-
-(rule
-  (targets bng_generic.c)
-  (deps bng_digit.c bng_amd64.c bng_arm64.c bng_ia32.c bng_ppc.c)
-  (action (run touch bng_generic.c)))
+  (flags -w +a-4-9-41-42-44-45-48 -warn-error +A -g -safe-string -strict-sequence -strict-formats))

--- a/test/dune
+++ b/test/dune
@@ -1,5 +1,5 @@
-(alias
-  (name runtest)
+(rule
+  (alias runtest)
   (action (progn (run %{dep:driver.bc}) (run %{dep:driver.exe}))))
 
 (library
@@ -11,6 +11,7 @@
   (name driver)
   (modules driver)
   (flags -linkall)
+  (modes (byte exe))
   (libraries test_lib))
 
 (rule (with-stdout-to driver.ml (echo "")))

--- a/test/dune
+++ b/test/dune
@@ -1,6 +1,10 @@
 (rule
-  (alias runtest)
-  (action (progn (run %{dep:driver.bc}) (run %{dep:driver.exe}))))
+ (alias runtest)
+ (action (run ./driver.exe)))
+
+(rule
+ (alias runtest)
+ (action (run ./driver.bc)))
 
 (library
   (name test_lib)

